linux/guest_customization/handle_gosc_known_issues.yml

```diff
@@ -0,0 +1,93 @@
+
+            - name: "Known issue - Perl GOSC of {{ vm_guest_os_distribution }} on vCenter Server {{ vcenter_version }}"
+              ansible.builtin.debug:
+                msg:
+                  - "Perl guest customization would fail due to not found /etc/sysconfig/network-scripts/ on {{ vm_guest_os_distribution }}."
+                  - "Please refer to https://issues.redhat.com/browse/RHEL-40365."
+                  - "Created the directory /etc/sysconfig/network-scripts as a workaround."
+              tags:
+                - known_issue
+
+    # Perl GOSC fails on Ubuntu 23.10 or later versions due to hwclock not built-in
+    - name: "Handle known issue for {{ vm_guest_os_distribution }} on vCenter Server {{ vcenter_version }}"
+      when:
+        - vcenter_version is version('8.0.3', '<')
+        - guest_os_ansible_distribution == "Ubuntu"
+        - guest_os_ansible_distribution_ver is version('23.10', '>=')
+      block:
+        - name: "Uninstall current cloud-init"
+          include_tasks: ../utils/install_uninstall_package.yml
+          vars:
+            package_list: ["util-linux-extra"]
+            package_state: "present"
+
+        - name: "Known issue - Perl GOSC of {{ vm_guest_os_distribution }} on vCenter Server {{ vcenter_version }}"
+          ansible.builtin.debug:
+            msg:
+              - "Perl guest customization would fail due to path to hwclock not found"
+              - "Please refer to https://knowledge.broadcom.com/external/article?legacyId=95091"
+              - "Install package 'util-linux-extra' as a workaround"
+          tags:
+            - known_issue
+
+# Cloud-init GOSC failure on Oracle Linux 8.7 and 9.1
+- name: "Handle known issue for {{ vm_guest_os_distribution }} on vCenter Server {{ vcenter_version }}"
+  when:
+    - enable_cloudinit_gosc | bool
+    - guest_os_ansible_distribution == "OracleLinux"
+    - guest_os_ansible_distribution_ver in ['8.7', '9.1']
+  block:
+    - name: "Uninstall current cloud-init"
+      include_tasks: ../utils/install_uninstall_package.yml
+      vars:
+        package_list: ["cloud-init"]
+        package_state: "absent"
+
+    - name: "Remove /etc/cloud dir before install the latest cloud-init"
+      ansible.builtin.shell: |
+        if [ -e "/etc/cloud" ] ; then
+          rm -rf "/etc/cloud";
+        fi
+      delegate_to: "{{ vm_guest_ip }}"
+
+    - name: "Install latest cloud-init version"
+      include_tasks: ../utils/install_uninstall_package.yml
+      vars:
+        package_list: ["cloud-init"]
+        package_state: "latest"
+
+    - name: "Known issue - failure of cloud-init GOSC in Oracle Linux 8.7 and 9.1"
+      ansible.builtin.debug:
+        msg:
+          - "Cloud-init guest customization would fail because cloud-init-local.service couldn't start when using cloud-init 22.1-5.0.1."
+          - "The issue has been resolved in cloud-init 22.1-6.0.1. In this case cloud-init is upgraded to latest version as a workaround."
```

Comment:
Like statements above, here can be changed:
In this case cloud-init is upgraded to latest version as a workaround. -> Upgrade cloud-init to the latest version as a workaround.